server/util/stream_reader.go

```diff
@@ -67,21 +67,30 @@ func (s *streamReader[T, U]) Run() error {
 
 func (s *streamReader[T, U]) handleServerStream() {
 	for {
-		if message, err := s.stream.Recv(); err != nil {
+		message, err := s.stream.Recv()
+		if err != nil {
 			s.close(err)
```

Comment:
We could put `s.close(err)` into a `defer` function here, because it's need to be done each time

Reply:
done, PTAL